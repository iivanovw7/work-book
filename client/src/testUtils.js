import { createMemoryHistory } from 'history';
import * as localized from './assets/locales.json';

// Mock browser history
export const history = createMemoryHistory('/dashboard');

// Grid config object for testing
export const mappedSizes = {
	xs: 12,
	sm: 12,
	md: 12,
	lg: 11
};

export const localizedText = {
	locales: localized.locales,
	topBar: localized.topBar,
	login: localized.login,
	navigation: localized.navigation,
	pageNotFound: localized.pageNotFound
};
