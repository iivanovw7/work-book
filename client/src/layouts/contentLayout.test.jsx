import React from 'react';
import 'cross-fetch/polyfill';
import 'jsdom-global/register';
import { mount } from 'enzyme';
import chalk from 'chalk';
import { MemoryRouter } from 'react-router-dom';
import ContentLayout from './contentLayout';

import * as testUtils from '../testUtils';

jest.mock('../config/apiURL', () => () => '/api');
jest.mock('../utils');

describe(`Testing [${chalk.yellow('content layout')}] component: `, () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  const props = {
    locale: 'eng',
    text: testUtils.localizedText,
    theme: 'dark',
    history: testUtils.history
  };

  const ChildComponent = () => (
    <div>Component</div>
  );

  const Composition = () => (
    <MemoryRouter>
      <ContentLayout userAccess {...props}>
        <ChildComponent />
      </ContentLayout>
    </MemoryRouter>
  );

  it('Should match snapshot', async () => {
    const component = mount(<Composition />);
    expect(component).toMatchSnapshot();
  });

  it('Should mount correctly with props', async () => {
    const component = mount(<Composition />);
    expect(component.find('div').text()).toBe('Component');
    expect(component.find({ theme: 'dark' }).length).toEqual(1);
  });
});
