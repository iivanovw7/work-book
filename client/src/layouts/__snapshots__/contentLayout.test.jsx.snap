// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing content layout component:  Should match snapshot 1`] = `ReactWrapper {}`;
